### PR TITLE
Silence Coverity issues

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/34)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/34#-:-Kw-ZCI57zLPIIsR0mpj:bnfp4nl)*

**Review by GBuella — APPROVED** [review]

File: test/test_syscalls.c [hunk 1]
```diff
@@ -201,7 +201,7 @@ s();
 s();
 	(void) lseek(fd, 0, SEEK_SET);
 s();
-	(void) read(fd, buffer, BUF_SIZE);
+	int rb = read(fd, buffer, BUF_SIZE);
```
Comment (marcinslusarz) [inline test/test_syscalls.c]: I don't see how it can affect Coverity...

File: src/utils.c [hunk 2]
```diff
@@ -323,7 +323,7 @@ str_replace_all(char **const text, const char *templt, const char *str)
 	char *occ;
 
 	if (str_len <= templt_len) {
-		char *new_str = malloc(templt_len);
+		char *new_str = malloc(templt_len + 1);
```
Comment (marcinslusarz) [inline src/utils.c]: What is the warning/error you are fixing here?

File: test/test_syscalls.c (hunk 1 above)
Comment (ldorau) [reply test/test_syscalls.c]: However, it does affect ...

File: src/utils.c (hunk 2 above)
Comment (ldorau) [reply src/utils.c]:
```
CID 180942 (#1 of 1): Out-of-bounds access (OVERRUN)
3. alloc_strlen: Allocating insufficient memory for the terminating null of the string.
```

File: test/test_syscalls.c (hunk 1 above)
Comment (ldorau) [reply test/test_syscalls.c]: It is not my fault that Coverity works like this ;-)

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/34#-:-Kw4P9g8w7Cw1C8mIlL2:bnfp4nl)*

**Comment by krzycz** [comment]
Reviewed 2 of 2 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/34)*

**Pull request merged**